activation/handler_v1.go

```diff
@@ -438,7 +438,6 @@ func (h *HandlerV1) checkWrongPrevAtx(
 		return nil, err
 	}
 	if v != types.AtxV1 {
-		// TODO(mafa): update when V2 is introduced
```

Comment:
What about this case? Don't you plan to implement this malfeasance proof eventually?

Reply:
A V1 ATX can't have a V2 ATX as prev ATX, because at least one of them is syntactically invalid. Either the V1 has been published in an epoch where only V2 are allowed or vice versa.

Reply:
But here, the `atx2ID` (`v` is the version of which) is **another** ATX that uses the same previous ATX as the processed ATX. AFAIR, this can happen. Consider a scenario below (ATXs are received and processed top-town, V2 is valid since the publish epoch 3). The ATX `C` is published "in the past" using the same previous ATX as a V2 ATX `B`.

| publish epoch | ATX published | version | previous ATX |
|-|-|-|-|
|1 | A | 1| NULL|
|3 | B | 2| A|
|2 | C | 1| A|

In this scenario, we would hit this condition with `atx2ID == B` when handling received ATX `C`.